internal/batch/batch_processor_test.go

```diff
@@ -88,22 +89,75 @@ func TestUnfilledBatch(t *testing.T) {
 
 	// Kick off a go routine to consume the confirmations
 	go func() {
-		for i := 0; i < 5; i++ {
+		for i := 0; i < len(work); i++ {
 			<-work[i].dispatched
 		}
 		wg.Done()
 	}()
 
 	// Dispatch the work
-	for i := 0; i < 5; i++ {
+	for i := 0; i < len(work); i++ {
 		bp.newWork <- work[i]
 	}
 
 	// Wait for the confirmations, and the dispatch
 	wg.Wait()
 
 	// Check we got all the messages in a single batch
-	assert.Equal(t, len(dispatched[0].Payload.Messages), 5)
+	assert.Equal(t, len(dispatched[0].Payload.Messages), len(work))
+
+	bp.close()
+	bp.waitClosed()
+
+}
+
+func TestBatchSizeOverflow(t *testing.T) {
+	log.SetLevel("debug")
+
+	wg := sync.WaitGroup{}
+	wg.Add(3)
+
+	dispatched := []*fftypes.Batch{}
+	mdi, bp := newTestBatchProcessor(func(c context.Context, b *fftypes.Batch, s []*fftypes.Bytes32) error {
+		dispatched = append(dispatched, b)
+		wg.Done()
+		return nil
+	})
+	bp.conf.BatchMaxBytes = 1
+	mockRunAsGroupPassthrough(mdi)
+	mdi.On("UpdateMessages", mock.Anything, mock.Anything, mock.Anything).Return(nil)
+	mdi.On("UpsertBatch", mock.Anything, mock.Anything, mock.Anything).Return(nil)
+	mdi.On("UpdateBatch", mock.Anything, mock.Anything).Return(nil)
+
+	// Generate the work the work
```

Comment:
work work

Reply:
⛏️ ⛏️

Reply:
nicely I had 4 instances of this comment - fixed now